File: game-template/tien-len-deno-ts/test/go/performance.go
package main

import (
	"fmt"
	"time"
)

const totalIterations = 1000000000 // Số lần lặp (1 tỷ)

func performHeavyTask() uint64 {
	var result uint64
	for i := uint64(0); i < totalIterations; i++ {
		result += i
	}
	return result
}

func main() {
	startTime := time.Now()
	result := performHeavyTask()
	endTime := time.Now()

	elapsedTime := endTime.Sub(startTime)

	fmt.Printf("Kết quả: %d\n", result)
	fmt.Printf("Thời gian thực thi: %v\n", elapsedTime)
}

File: game-template/tien-len-deno-ts/test/performance.rs
use std::time::Instant;

const TOTAL_ITERATIONS: u64 = 1_000_000_000; // Số lần lặp (1 tỷ)

fn perform_heavy_task() -> u64 {
    let mut result: u64 = 0;
    for i in 0..TOTAL_ITERATIONS {
        result += i;
    }
    result
}

fn main() {
    let start_time = Instant::now();
    let result = perform_heavy_task();
    let elapsed_time = start_time.elapsed();

    println!("Kết quả: {}", result);
    println!("Thời gian thực thi: {:?}", elapsed_time);
}
